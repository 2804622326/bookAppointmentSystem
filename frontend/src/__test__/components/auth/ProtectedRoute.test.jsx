import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProtectedRoute from '../../../components/auth/ProtectedRoute.jsx';

jest.mock('react-router-dom', () => ({
  Navigate: ({ to }) => <div>redirect {to}</div>,
  useLocation: () => ({ pathname: '/private' }),
}));

test('redirects to login if not authenticated', () => {
  localStorage.removeItem('authToken');
  const { asFragment } = render(
    <ProtectedRoute allowedRoles={['PATIENT']}>
      <div>secret</div>
    </ProtectedRoute>
  );
  expect(screen.getByText('redirect /login')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('redirects to unauthorized if role not allowed', () => {
  localStorage.setItem('authToken', 't');
  localStorage.setItem('userRoles', JSON.stringify(['OTHER']));
  const { asFragment } = render(
    <ProtectedRoute allowedRoles={['PATIENT']}>
      <div>secret</div>
    </ProtectedRoute>
  );
  expect(screen.getByText('redirect /unauthorized')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('renders children when authorized', () => {
  localStorage.setItem('authToken', 't');
  localStorage.setItem('userRoles', JSON.stringify(['PATIENT']));
  const { asFragment } = render(
    <ProtectedRoute allowedRoles={['PATIENT']}>
      <div>secret</div>
    </ProtectedRoute>
  );
  expect(screen.getByText('secret')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
