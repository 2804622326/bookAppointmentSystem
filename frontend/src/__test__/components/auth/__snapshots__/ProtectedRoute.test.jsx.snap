// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects to login if not authenticated 1`] = `
<DocumentFragment>
  <div>
    redirect /login
  </div>
</DocumentFragment>
`;

exports[`redirects to unauthorized if role not allowed 1`] = `
<DocumentFragment>
  <div>
    redirect /unauthorized
  </div>
</DocumentFragment>
`;

exports[`renders children when authorized 1`] = `
<DocumentFragment>
  <div>
    secret
  </div>
</DocumentFragment>
`;
